specs/phase0/p2p-interface.md

```diff
@@ -310,6 +315,8 @@ The following validations MUST pass before forwarding the `signed_aggregate_and_
   (via aggregate gossip, within a verified block, or through the creation of an equivalent aggregate locally).
 - _[IGNORE]_ The `aggregate` is the first valid aggregate received for the aggregator
   with index `aggregate_and_proof.aggregator_index` for the epoch `aggregate.data.target.epoch`.
+- _[IGNORE]]_ The block being voted for (`aggregate.data.beacon_block_root`) has been seen locally
```

Comment:
The message ID and seen-cache already does this for us. Except when the cache overflows (It's a large LRU in Rust and Go at least), and the old entry may disappear. Probably fine to keep it here, hope it's not too difficult to have a 2nd "has this been seen" lookup for clients.

Reply:
there's a chance you didn't see the block via gossip though. Maybe retrieved in a blocks-by-root request triggered by something else

Reply:
Good point about sync and other sources. However, as discussed offline, restricting it to forkchoice contents may miss other edge-case seen-block cases. Changing it to something like "any seen blocks (including from non-gossip sources)" should work.